src/api/billing.ts

```diff
@@ -0,0 +1,144 @@
+import { openmrsFetch } from "@openmrs/esm-framework";
+import { ENDPOINTS, CONFIG } from "../constants";
+
+/**
+ * Submit billing information after form submission
+ *
+ * @param payload - The data to submit
+ * @returns Response from the billing endpoint
+ */
+export async function submitBillingData(payload: any) {
+  return openmrsFetch(ENDPOINTS.AFTER_POST_SUBMISSION, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(payload),
+  });
+}
+
+/**
+ * Gets global bills for a patient
+ *
+ * @param patientUuid - The patient's UUID
+ * @returns List of global bills
+ */
+export async function getGlobalBills(patientUuid: string) {
+  const response = await openmrsFetch(
+    `${ENDPOINTS.GLOBAL_BILL}?patient=${patientUuid}&v=full`
+  );
+  return response.data?.results || [];
+}
+
+/**
+ * Gets insurance policies for a patient
+ *
+ * @param patientUuid - The patient's UUID
+ * @returns List of insurance policies
+ */
+export async function getInsurancePolicies(patientUuid: string) {
+  const response = await openmrsFetch(
+    `${ENDPOINTS.INSURANCE_POLICY}?patient=${patientUuid}&v=full`
+  );
+  return response.data?.results || [];
+}
+
+/**
+ * Gets the insurance number concept UUID from system settings
+ *
+ * @returns Insurance number concept UUID
+ */
+export async function getInsuranceNumberConceptUuid() {
+  const response = await openmrsFetch(
+    `${ENDPOINTS.SYSTEM_SETTING}?q=${CONFIG.INSURANCE_NUMBER_CONCEPT_SETTING}&v=custom:(value)`
+  );
+  return response.data?.results?.[0]?.value;
+}
+
+/**
+ * Gets observations for a patient with a specific concept
+ *
+ * @param patientUuid - The patient's UUID
+ * @param conceptUuid - The concept UUID to filter by
+ * @returns List of observations
+ */
+export async function getPatientObsByConceptUuid(
+  patientUuid: string,
+  conceptUuid: string
+) {
+  const response = await openmrsFetch(
+    `${ENDPOINTS.OBS}?patient=${patientUuid}&concept=${conceptUuid}&v=custom:(valueText)&limit=1`
+  );
+  return response.data?.results || [];
+}
+
+/**
+ * Gets insurance policy number for a patient using multiple strategies
+ *
+ * @param patientUuid - The patient's UUID
+ * @returns The insurance policy number
+ */
+export async function getInsurancePolicyNumber(
+  patientUuid: string
+): Promise<string> {
+  try {
+    const globalBills = await getGlobalBills(patientUuid);
+
+    if (globalBills.length > 0) {
+      const openGlobalBill = globalBills.find((bill) => bill.closed === false);
+      const globalBill = openGlobalBill || globalBills[0];
+
+      if (globalBill.billIdentifier) {
+        const insuranceCardNo = globalBill.billIdentifier.split(/\d+/)[0];
```

Comment:
I would avoid splitting this, it's prone to error. 
Instead:
globalBill has a many-to-one relationship to Admission. 
Admission has a many-to-one relationship to InsurancePolicy
InsurancePolicy has insuranceCardNo.
so --  globalBill.admission.insurancePolicy.insuranceCardNo should take precedence. It's more exact